internal/storage/local_chunk_manager.go

```diff
@@ -18,82 +18,145 @@ package storage
+func NewLocalChunkManager(opts ...Option) *LocalChunkManager {
+	c := newDefaultConfig()
+	for _, opt := range opts {
+		opt(c)
+	}
 	return &LocalChunkManager{
-		localPath: localPath,
+		localPath: c.rootPath,
 	}
 }
 
 // GetPath returns the path of local data if exists.
-func (lcm *LocalChunkManager) GetPath(key string) (string, error) {
-	if !lcm.Exist(key) {
-		return "", errors.New("local file cannot be found with key:" + key)
+func (lcm *LocalChunkManager) GetPath(filePath string) (string, error) {
+	if !lcm.Exist(filePath) {
+		return "", errors.New("local file cannot be found with filePath:" + filePath)
 	}
-	path := path.Join(lcm.localPath, key)
-	return path, nil
+	absPath := path.Join(lcm.localPath, filePath)
+	return absPath, nil
 }
 
 // Write writes the data to local storage.
-func (lcm *LocalChunkManager) Write(key string, content []byte) error {
-	filePath := path.Join(lcm.localPath, key)
-	dir := path.Dir(filePath)
+func (lcm *LocalChunkManager) Write(filePath string, content []byte) error {
+	absPath := path.Join(lcm.localPath, filePath)
+	dir := path.Dir(absPath)
 	if _, err := os.Stat(dir); os.IsNotExist(err) {
 		err := os.MkdirAll(dir, os.ModePerm)
 		if err != nil {
 			return err
 		}
 	}
-	err := ioutil.WriteFile(filePath, content, 0600)
+	err := ioutil.WriteFile(absPath, content, os.ModePerm)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
+// MultiWrite writes the data to local storage.
+func (lcm *LocalChunkManager) MultiWrite(contents map[string][]byte) error {
+	var el errorutil.ErrorList
+	for filePath, content := range contents {
+		err := lcm.Write(filePath, content)
+		if err != nil {
+			el = append(el, err)
+		}
+	}
+	if len(el) == 0 {
+		return nil
+	}
+	return el
+}
+
 // Exist checks whether chunk is saved to local storage.
-func (lcm *LocalChunkManager) Exist(key string) bool {
-	path := path.Join(lcm.localPath, key)
-	_, err := os.Stat(path)
-	if err != nil {
-		return os.IsExist(err)
+func (lcm *LocalChunkManager) Exist(filePath string) bool {
+	absPath := path.Join(lcm.localPath, filePath)
+	if _, err := os.Stat(absPath); errors.Is(err, os.ErrNotExist) {
+		return false
 	}
 	return true
 }
 
 // Read reads the local storage data if exists.
-func (lcm *LocalChunkManager) Read(key string) ([]byte, error) {
-	filePath := path.Join(lcm.localPath, key)
-	file, err := os.Open(path.Clean(filePath))
+func (lcm *LocalChunkManager) Read(filePath string) ([]byte, error) {
+	if !lcm.Exist(filePath) {
+		return nil, errors.New("file not exist" + filePath)
+	}
+	absPath := path.Join(lcm.localPath, filePath)
+	file, err := os.Open(path.Clean(absPath))
 	if err != nil {
 		return nil, err
 	}
-	defer file.Close()
 	content, err := ioutil.ReadAll(file)
 	if err != nil {
 		return nil, err
 	}
-	return content, nil
+	return content, file.Close()
+}
+
+// MultiRead reads the local storage data if exists.
+func (lcm *LocalChunkManager) MultiRead(filePaths []string) ([][]byte, error) {
+	var results [][]byte
```

Comment:
results := make([][]byte, len(filePaths))
always pre alloacate slice length if know it